_posts/blog/2016-07-18-subjects.md

```diff
@@ -0,0 +1,142 @@
+>*I would not advise using Observable.Create() as Subject is much easier to use and good enough 99% of time*. 
+
+Since then I’ve learnt that they are just simple tools to start an ``Observable`` stream and by themselves tools are neither good nor evil, it just depends how we use them. 
+
+## The ease of usage...
+
+The best thing about ``Subjects`` is how easy they are to use. You can simply push a value in one place and use it somewhere else as a normal ``Observable``:
+
+{% highlight java %}
+PublishSubject<String> subject = PublishSubject.create();
+subject.subscribe(text -> Log.d(“Subjects:”, text));
+subject.onNext(“Easy to use”);
+{% endhighlight %}
+
+This is because ``Subjects`` extend ``Observable`` and implement ``Observer``. 
+
+Thanks to being an ``Observable``, we can use all the typical operators (``map``, ``flatMap``, etc) without any extra hassle:
+
+{% highlight java %}
+subject.filter(str -> !str.isEmpty())
+.map(str -> “This is a not empty string: ” + str);
+{% endhighlight %}
+
+The benefit of being an ``Observer`` is that we can directly put any value to ``onNext``, throw an exception in ``onError`` or just complete. 
+
+{% highlight java %}
+subject.onNext(“Some value”);
+subject.onComplete();
+{% endhighlight %}
+
+## ... comes with a price
+
+Some say that ``Subjects`` are broken, but I would disagree. The creators of ``Subjects`` did a great job, they've focused on creating a user-friendly tool, despite some limitations.
+
+With ``Subjects``, a great power comes with a great burden.
+
+### Backpressure
+
+Basically, backpressure exception occurs when an operator produces values faster than you can handle further downstream. 
+
+I like to compare operators that can handle backpressure to waiters. If clients of a restaurant are full, then they would tell the waiter to stop bringing food and naturally, no more food will be served. On the other hand, operators that cannot handle backpressure are like grandmas. If you tell her that you cannot eat any more, she will just ignore you and keep on feeding you more food until you explode or run away. 
+
+
+The issue with ``Subjects`` and backpressure comes from the fact that ``Subjects`` do not extend ``Subscriber``. As <a href="https://twitter.com/akarnokd">Dávid Karnok</a> mentioned in his blog, ``Subscriber`` is the main way to handle backpressure in RxJava. Unfortunately, Java does not have multiple inheritance. The designers have chosen to make ``Subjects`` simpler to use rather than support backpressure. 
+
+There is really simple test to check if your ``Observable`` supports backpressure. The trick is to request via ``TestSubscriber`` less elements that we put into the ``Subject``:
+
+{% highlight java %}
+@Test
+public void check_backpressure_support() {
+  TestSubscriber<Boolean> ts = new TestSubscriber<>(**1**); // The Subscriber requests only 1 value
```

Comment:
are you trying to make 1 bold? it doesn't work in the code area. 
Also, put the comment about the subscriber either above or below it. otherwise you need to scroll left